lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -228,32 +235,44 @@ def self.upload_binary(app_id, binary_path)
         connection.post("/app-binary-uploads?app_id=#{app_id}", File.open(binary_path).read) do |request|
           request.headers["Authorization"] = "Bearer " + auth_token
         end
+      rescue Faraday::ResourceNotFound
+        UI.crash!(ErrorMessage::INVALID_APP_ID)
+      rescue Errno::ENOENT
+        UI.crash!(ErrorMessage::APK_NOT_FOUND)
       end
 
       def self.upload(app_id, binary_path)
         upload_token = get_upload_token(app_id, binary_path)
-        status = upload_status(upload_token, app_id)
+        status = upload_status(app_id, upload_token)
         if status == "SUCCESS"
           UI.message("This APK/IPA has been uploaded before. Skipping upload step.")
         else
+          UI.message("This APK has not been uploaded before.")
           MAX_POLLING_RETRIES.times do
             if status == "SUCCESS"
               UI.message("Uploaded APK/IPA Successfully!")
               break
             elsif status == "IN_PROGRESS"
-              sleep(POLLING_INTERVAL_S)
+              sleep(POLLING_INTERVAL_SECONDS)
             else
-              UI.message("Uploading the APK/IPA")
+              UI.message("Uploading the APK/IPA.")
               upload_binary(app_id, binary_path)
             end
-            status = upload_status(upload_token, app_id)
+            status = upload_status(app_id, upload_token)
+          end
+          if status != "SUCCESS"
+            UI.message("It took longer than expected to process your APK/IPA, please try again")
           end
         end
       end
 
-      def self.upload_status(app_token, app_id)
-        response = connection.get("#{PATH}#{app_id}/upload_status/#{app_token}") do |request|
-          request.headers["Authorization"] = "Bearer " + auth_token
+      def self.upload_status(app_id, app_token)
+        begin
+          response = connection.get("#{PATH}#{app_id}/upload_status/#{app_token}") do |request|
+            request.headers["Authorization"] = "Bearer " + auth_token
+          end
+        rescue Faraday::ResourceNotFound
+          UI.crash!(ErrorMessage::INVALID_APP_ID)
```

Comment:
Ditto to earlier suggestion about giving the failing value in the error message.